java/ingest/ingest-runner/src/main/java/sleeper/ingest/impl/IngesterIntoPartitions.java

```diff
@@ -168,4 +182,42 @@ public CompletableFuture<List<FileReference>> initiateIngest(
                         .map(CompletableFuture::join)
                         .collect(Collectors.toList()));
     }
+
+    public CompletableFuture<List<FileReference>> ingestOneFileWithReferencesInLeafPartitions(
+            CloseableIterator<Record> orderedRecordIterator, PartitionTree partitionTree) throws IOException {
+        List<String> rowKeyNames = sleeperSchema.getRowKeyFieldNames();
+        Map<String, Long> partitionIdToRecordCount = new HashMap<>();
+        Partition currentPartition = null;
+        List<CompletableFuture<FileReference>> completableFutures = new ArrayList<>();
+        PartitionFileWriter rootFileWriter = partitionFileWriterFactoryFn.apply(partitionTree.getRootPartition());
+        try {
+            while (orderedRecordIterator.hasNext()) {
+                Record record = orderedRecordIterator.next();
+                rootFileWriter.append(record);
+                Key key = Key.create(record.getValues(rowKeyNames));
+                if (currentPartition == null || !currentPartition.isRowKeyInPartition(sleeperSchema, key)) {
+                    currentPartition = partitionTree.getLeafPartition(sleeperSchema, key);
+                }
+                partitionIdToRecordCount.compute(currentPartition.getId(),
+                        (partitionId, recordCount) -> (recordCount == null) ? 1 : recordCount + 1);
+            }
+            completableFutures.add(rootFileWriter.close());
```

Comment:
It looks like this is the only CompletableFuture, and the list can be removed.

Reply:
Done, thanks